src/kernel/include/cap_ops.h

```diff
@@ -80,8 +80,9 @@ cap_kmem_activate(struct captbl *t, capid_t cap, unsigned long addr, unsigned lo
 	pgtblc = (struct cap_pgtbl *)captbl_lkup(t, cap);
 
 	if (unlikely(!pgtblc)) return -ENOENT;
-	if (unlikely(pgtblc->h.type != CAP_PGTBL || pgtblc->lvl != 0)) return -EINVAL;
-	if ((ret = pgtbl_kmem_act(pgtblc->pgtbl, addr & PGTBL_FRAME_MASK, (unsigned long *)kern_addr, pte_ret))) return ret;
+	/* PRY:some modification here */
+	if (unlikely(pgtblc->h.type != CAP_PGTBL /* || pgtblc->lvl != 0*/)) return -EINVAL;
```

Comment:
The root level is not indicated with lvl now. instead, it is now indicated in another field.